phoenix-core/src/main/java/org/apache/phoenix/compile/WhereOptimizer.java

```diff
@@ -114,6 +116,7 @@ public static Expression pushKeyExpressionsToScan(StatementContext context, Filt
     	boolean isMultiTenant = tenantId != null && table.isMultiTenant();
     	if (isMultiTenant) {
     		tenantId = ScanUtil.padTenantIdIfNecessary(schema, isSalted, tenantId);
```

Comment:
Take a look at ScanUtil.padTenantIdIfNecessary(), as it assumes tenantId is a string:
    public static PName padTenantIdIfNecessary(RowKeySchema schema, boolean isSalted, PName tenantId) {
        int pkPos = isSalted ? 1 : 0;
        String tenantIdStr = tenantId.getString();
        Field field = schema.getField(pkPos);
        PDataType dataType = field.getDataType();
        boolean isFixedWidth = dataType.isFixedWidth();
        Integer maxLength = field.getMaxLength();
        if (isFixedWidth && maxLength != null) {
            if (tenantIdStr.length() < maxLength) {
                tenantIdStr = (String)dataType.pad(tenantIdStr, maxLength);
                return PNameFactory.newName(tenantIdStr);
            }
        }
        return tenantId;
    }

Probably best if your ScanUtil.getTenantIdBytes() function takes into account padding as well and we can get rid of this util function (see comment below on that).